crates/project/src/image_store.rs

```diff
@@ -143,9 +221,31 @@ impl ProjectItem for ImageItem {
         // Since we do not have a way to toggle to an editor
         if Img::extensions().contains(&ext) && !ext.contains("svg") {
             Some(cx.spawn(|mut cx| async move {
-                project
+                let image_model = project
                     .update(&mut cx, |project, cx| project.open_image(path, cx))?
-                    .await
+                    .await?;
+
+                let fs = Arc::new(fs::RealFs::default());
+                let project_clone = project.clone();
+
+                if let Ok(()) = image_model.update(&mut cx, |image, cx| {
+                    let project_ref = project_clone.read(cx);
+
+                    if let Ok(metadata) = futures::executor::block_on(image.load_metadata(
```

Comment:
This shouldn't be blocked on. Rather, return it out of the `.update()`, then `.await` it, then call `update()` again to assign the results.

Reply:
oh! thanks, i'll try this now

Reply:
@mikayla-maki, I did some cleanups. Kindly review and let me know what you think. 

The only issue I have is in this block. When I try to do it like you said &mdash; regarding this `executor::block_on` &mdash; I run into Rust lifetime issues several times.